DevSkim-DotNet/Microsoft.DevSkim.CLI/Commands/SuppressionCommand.cs

```diff
@@ -0,0 +1,126 @@
+        }
+
+        public int Run()
+        {
+            var sarifLog = SarifLog.Load(_opts.SarifInput);
+            if (sarifLog.Runs.Count > 0)
+            {
+                var run = sarifLog.Runs[0];
+                var groupedResults = run.Results.GroupBy(x => x.Locations[0].PhysicalLocation.ArtifactLocation.Uri);
+                if (!_opts.ApplyAllSuppression && !_opts.FilesToApplyTo.Any() && !_opts.RulesToApplyFrom.Any())
+                {
+                    Console.WriteLine("Must specify either apply all suppression comments or a combination of file and rules to apply");
+                    return (int)ExitCode.CriticalError;
+                }
+
+                if (_opts.FilesToApplyTo.Any())
+                {
+                    groupedResults =
+                        groupedResults.Where(x => _opts.FilesToApplyTo.Any(y => x.Key.AbsolutePath.Contains(y)));
+                }
+
+                if (_opts.RulesToApplyFrom.Any())
+                {
+                    groupedResults = groupedResults.Where(x =>
+                        _opts.RulesToApplyFrom.Any(y =>
+                            x.Any(z =>
+                                z.RuleId == y)));
+                }
+
+                groupedResults = groupedResults.ToList();
+                foreach (var resultGroup in groupedResults)
+                {
+                    var fileName = resultGroup.Key;
+                    var potentialPath = Path.Combine(_opts.Path, fileName.OriginalString);
+                    // Flatten all the replacements into a single list
+                    var listOfReplacements = resultGroup
+                      .Where(x => x.Locations is { })
+                      .SelectMany(x => x.Locations).Select(x => x.PhysicalLocation).ToList();
+
+                    listOfReplacements
+                    .Sort((a, b) => a.Region.StartLine - b.Region.StartLine);
+
+                    var distinctReplacements  = listOfReplacements
+                    .GroupBy(x => x.Region.StartLine)
+                    .Select(x => x.FirstOrDefault());
+
+                    var ruleIds = string.Join(",", resultGroup.Select(x => x.RuleId).Distinct());
+
+                    if (File.Exists(potentialPath))
+                    {
+                        var theContent = File.ReadAllText(potentialPath).Split(Environment.NewLine);
+                        int currLine = 0;
+                        var sb = new StringBuilder();
+                        foreach (var replacement in distinctReplacements)
+                        {
+                            var zbStartLine = theContent[0] == string.Empty ? replacement.Region.StartLine: replacement.Region.StartLine - 1;
+                            var isMultiline = theContent[zbStartLine].EndsWith(@"\");
+                            var ignoreComment = $"{getPrefixComment(replacement.Region.SourceLanguage)} DevSkim: ignore {ruleIds} {getSuffixComment(replacement.Region.SourceLanguage)}";
```

Comment:
The ruleid should be drawn from the specific issue identified on the line in question